net.go

```diff
@@ -425,6 +426,122 @@ func (e *EurekaConnection) ScheduleSecureVIPAddressUpdates(addr string, await bo
 	return e.scheduleVIPAddressUpdates(addr, await, done, options), nil
 }
 
+// An InstanceSetSource holds a periodically updated set of instances registered with Eureka.
+type InstanceSetSource struct {
+	m         *sync.RWMutex
+	instances []*Instance
+	done      chan<- struct{}
+}
+
+func newInstanceSetSourceFromChannel(await bool, updates <-chan InstanceSetUpdate, done chan<- struct{}) *InstanceSetSource {
```

Comment:
any thoughts of combining secure and non secure functions to one with `secure : bool` parameter?